invokeai/frontend/web/src/features/metadata/util/recallers.ts

```diff
@@ -94,6 +97,13 @@ const recallScheduler: MetadataRecallFunc<ParameterScheduler> = (scheduler) => {
   getStore().dispatch(setScheduler(scheduler));
 };
 
+const recallInitialImage: MetadataRecallFunc<ParameterInitialImage> = (initialImage) => {
+  const image = {
+    image_name: initialImage,
+  };
+  getStore().dispatch(initialImageChanged(image as ImageDTO));
+};
+
```

Comment:
This is going to cause problems. Here's the reducer for that action:
```ts
initialImageChanged: (state, action: PayloadAction<ImageDTO>) => {
  const { image_name, width, height } = action.payload;
  state.initialImage = { imageName: image_name, width, height };
},
```

Elsewhere in the app, we'll be attempting to use the width and height but get undefined for them.

In this case, we should initiate a network request to retrieve the actual image DTO object. The metadata functions are typed to be async to support this. You'll need to use the imperative API for RTK Query:

```ts
// This will use the cache entry for the image if it exists, else hit the endpoint
const imageDTORequest = dispatch(imagesApi.endpoints.getImageDTO.initiate(image_name));
const imageDTO = await imageDTORequest.unwrap();
imageDTORequest.unsubscribe(); // make sure to unsubscribe to prevent memory leaks
```

Reply:
I started doing this at first but felt weird adding all the extra logic when the other recallers are short. Missed that it broke with deleted images, but tested and it works this way. Thanks! :)